src/core/debt_register.sol

```diff
@@ -173,4 +173,11 @@ contract DebtRegister is DSNote {
     function calcDebt(uint index, uint debtBalance_) private view returns (uint) {
         return rmul(debtBalance_, index);
     }
+
+    function rateSwitch(uint loan, uint currentRate, uint newRate) public auth {
+        require(now == rates[currentRate].lastUpdated);
+        require(now == rates[newRate].lastUpdated);
+        uint debt = calcDebt(rates[currentRate].index, debtBalance[loan]);
+        debtBalance[loan] = calcDebtBalance(rates[newRate].index, debt);
```

Comment:
don't you also have to decrease the debt of the old fee group and increase it for the new fee group?

Reply:
good catch. 🥇  I had in my head but I obviously forgot